components/resizable_sidebar/resizable_rhs/index.tsx

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {HTMLAttributes, useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/common';
+
+import {getRhsSize} from 'selectors/rhs';
+import LocalStorageStore from 'stores/local_storage_store';
+import {RHS_MIN_MAX_WIDTH, SidebarSize} from 'utils/constants';
+import {isOverLimit, isResizableSize, requestAnimationFrameForMouseMove, shouldRhsOverlapChannelView, shouldSnapWhenSizeGrown, shouldSnapWhenSizeShrunk} from '../utils';
+
+interface Props extends HTMLAttributes<'div'> {
+    children: React.ReactNode;
+}
+
+function Resizable({
```

Comment:
Would it be possible to share more of this code between the LHS and RHS? It would be nice to have a more generic `Resizable` component that we might be able to use in the future elsewhere, but that's certainly not necessary if it ends up being too difficult. I'd rather have two pieces of similar code than one full of a ton of branching logic that's hard to read.

From what I can see, the biggest differences between the two code-wise are where the default/max/min values come from, where the actual width is stored, and the extra overlapping logic for the RHS. Also, there's the fact that you're dragging from the left or right of the sidebar, but I don't know how much the code actually cares about that

Reply:
I tried to make a base `Resizable` component.  9e5ea4cfbd7f0a68ef7b445ebfe7e199f49879c0

Reply:
Awesome, thanks. I'll take a look at the component shortly